### PR TITLE
Add validation when base_url method is empty

### PR DESCRIPTION
**Pull request by blacksam07**
- Add validation when `base_url` method is empty
- Fix Rubocop offences

## Issue reference
#38

### CODE REVIEW

**Comment by nicozar95** [comment]
Great work! 🎉 
Could you add some specs?

**Comment by blacksam07** [comment]
> Great work! 🎉
> Could you add some specs?

Yes, in the next days

File: spec/wor/test_services/some_service_failure_spec.rb [hunk 1]
```diff
@@ -110,6 +110,18 @@
...
+
+      context 'when base_url malformed defined' do
+        subject(:service) { described_class.new(base_url_empty: true) }
+
+        before do
+          Spy.spy(service.logger, 'info')
+        end
+
+        it 'returns error with developer message' do
+          expect { service.method_with_log_with_block }.to raise_error
```
Comment (nicozar95) [inline spec/wor/test_services/some_service_failure_spec.rb]: Just a comment, some times the linter complains about the explicity of the error, in this case you should declared that is a `NoMethodError`.
But is OK!

**Review by nicozar95 — APPROVED** [review]

File: spec/wor/test_services/some_service_failure_spec.rb (hunk 1 above)
Comment (blacksam07) [reply spec/wor/test_services/some_service_failure_spec.rb]: In this rubocop version there is no problem, but in the future I want to upgrade rubocop to improve this and others

**Review by mtejedorwolox — APPROVED** [review]

File: lib/wor/requests/base.rb [hunk 2]
```diff
@@ -85,35 +87,39 @@ def uri(path)
       end
 
       def formatted_base_url
+        raise NoMethodError, 'The method base_url is malformed' if base_url.nil?
```
Comment (lucasVoboril) [inline lib/wor/requests/base.rb]: What do you think about to raise [a CustomError](https://github.com/Wolox/wor-requests/blob/master/lib/wor/requests/request_error.rb) here
```suggestion
        raise MalformedBaseUrl if base_url.nil?
```

```rb
  # wor-requests/lib/wor/requests/request_error.rb 

  class MalformedBaseUrl < StandardError
    def message
      'The method base_url is malformed'
    end
  end
```

File: spec/wor/test_services/some_service.rb [hunk 3]
```diff
@@ -71,6 +77,6 @@ def method_without_rescue
   protected
 
   def base_url
-    'https://baseurl.com'
+    'https://baseurl.com' unless base_url_empty
```
Comment (lucasVoboril) [inline spec/wor/test_services/some_service.rb]:
```suggestion
    base_url_empty || 'https://baseurl.com' 
```

Comment (blacksam07) [reply spec/wor/test_services/some_service.rb]: if use this the method `base_url ` return a boolean, and i can't testing when is empty

**Review by lucasVoboril — APPROVED** [review]

**Pull request merged**